### PR TITLE
fixes #225 - Added MultiLinePlotter app

### LINKED ISSUE #225: Add multi line plotting app for comparing number of tweets distribution over time
Add an app that will be able to visually compare number of tweets containing given keywords.

## Features

* user will input a keyword and press search
* the app will visualize the distribution of number of tweets containing the keyword over a period of time.
* the user can continue giving such keywords (one at a time), the app will plot a series in the same graph for each of the keywords).
* user will be able to compare the different distributions visually rather than comparing raw data.
* app will keep a record of the key words searched in a given session.
* app will also provide the tweets containing the given keywords.
user will also be able to remove a series from the graph.

### PR DESCRIPTION
**Pull request by djmgit**
fixes issue #225. Added MultiLinePlotter app
The app is able to compare tweet distribution, containing user
provided keywords, over a certain period of time.
User has to provide a keyword in the input field and
press '+' button. The tweet distribution will be
visualised using a line plot. User can continue providing
such words and multiple series will be plotted thus enabling the
users to compare them visually.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by djmgit** [comment]
Test link : http://loklakapps.surge.sh/MultiLinePlotter/

app1

**Comment by djmgit** [comment]
The app is not ready for store listing, will be doing it in next iteration.
Also the navbar height is different from that of the main page. The main page navbar may under go change in and after https://github.com/fossasia/apps.loklak.org/pull/214, so will fix the navbar height afetr that PR is merged.

**Comment by djmgit** [comment]
@kavithaenair @SKrPl @Achint08 @hemantjadon  open for review

File: MultiLinePlotter/css/styles.css [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    color: #C8254D;
+}
+
+.main {
+    margin-left: 35px;
+    margin-right: 35px;
+}
+
+.input-and-stat {
+
```
Comment (kavithaenair) [inline MultiLinePlotter/css/styles.css]: why to provide empty one, if no styling is required here?

File: MultiLinePlotter/css/styles.css [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+.tweet-reset:focus {
+    color: #FFFFFF;
+}
+
+.tweet-input-field:focus {
+  border-color: #D81B60;
+}
+
+#graph {
+
```
Comment (kavithaenair) [inline MultiLinePlotter/css/styles.css]: Here too

File: MultiLinePlotter/css/styles.css [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+.error:hover {
+    color: #c62828;
+    transition: all 0.5s;
+}
+
+.stat-body {
+    overflow: auto;
+    height: 190px;
+
```
Comment (kavithaenair) [inline MultiLinePlotter/css/styles.css]: Remove newline here

File: MultiLinePlotter/index.html [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      <div class="container-fluid navbar-container">
+        <div class="navbar-header">
+          <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#myNavbar">
+            <span class="icon-bar"></span>
+            <span class="icon-bar"></span>
+            <span class="icon-bar"></span>
+          </button>
+          <div class="navbar-brand brand-container">
+            <a href="http://apps.loklak.org" class="loklak-brand"> <span class="loklak-header">
+              loklak</span> <span>MultiLinePloter</span></a>
```
Comment (kavithaenair) [inline MultiLinePlotter/index.html]: Typo

File: MultiLinePlotter/index.html [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          </div>
+        </div>
+        <div class="collapse navbar-collapse" id="myNavbar">
+          <ul class="nav navbar-nav navbar-right">
+            <li><a href="http://loklak.net/about">About</a></li>
+            <li><a href="http://loklak.net/">Search</a></li>
+            <li><a href="http://blog.loklak.net/">Blog</a></li>
+            <li><a href="http://dev.loklak.org/">Developers</a></li>
+            <li><a href="http://api.loklak.org/">Api</a></li>
```
Comment (kavithaenair) [inline MultiLinePlotter/index.html]: Change it to `API`

File: MultiLinePlotter/css/styles.css [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+.stat-item {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    align-items: center;
+    justify-content: space-between;
+    transition: 0.5s;
```
Comment (Achint08) [inline MultiLinePlotter/css/styles.css]: Which CSS are you doing transition for?

File: MultiLinePlotter/css/styles.css [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+}
+
+.error {
+    color: #e57373;
+    border-radius: 4px;
+    border-style: solid;
+    border-width: 0.1px;
+    margin-top: 10px;
+    margin-right: 10px;
```
Comment (Achint08) [inline MultiLinePlotter/css/styles.css]: Instead of using `margin-*`. Use just one margin property and specify for all the edges.

**Comment by hemantjadon** [comment]
@Achint08 What did you found out? Can you open an issue for the same.

File: MultiLinePlotter/css/styles.css (hunk 1 above)
Comment (djmgit) [inline MultiLinePlotter/css/styles.css]: my bad, I will remove those

File: MultiLinePlotter/index.html (hunk 5 above)
Comment (djmgit) [inline MultiLinePlotter/index.html]: sure

**Comment by djmgit** [comment]
@kavithaenair @Achint08 @hemantjadon did the changes :)

**Comment by Achint08** [comment]
@hemantjadon The issue is that we need to specify a date for the correct aggregations.Currently, we have not specified the `date`. Moreover, we have kept `limit=10`. We are requesting for aggregations for last 10 days. Moreover, I think there is some issue with chart module. It shows up the tweets frequency wrong if you tally up. I will be opening up an issue.

**Comment by Achint08** [comment]
@djmgit Just handle the errors properly to avoid your application to crash.
This is the error I just came across.
screen shot 2017-06-30 at 5 00 48 pm.
Just look out for this in this PR or maybe next PR.

**Comment by djmgit** [comment]
Test link : http://loklakapps.surge.sh/MultiLinePlotter/

@Achint08 I am unable to reproduce this error, please check the app once again, if you see a different UI then I guess I had hosted the wrong branch previously. I have re-hosted the app

**Comment by kavithaenair** [comment]
@djmgit do we really need the top navbar, as i can see no other app having such top navbar? Then there will be no uniformity among apps.

**Comment by djmgit** [comment]
@kavithaenair some of the apps like sentiment visualiser, tweetfeed, whatsTrending, buble charts, user profile page, query browser, LQL and all the account apps hav navbar although the number is few.
So I am not sure whether I should keept the navbar or not.
@Achint08 @hemantjadon @SKrPl what do you guys suggest?

**Comment by kavithaenair** [comment]
@djmgit top navbar as in apps.loklak.org. I was pointing the navbar menu actually.

**Comment by djmgit** [comment]
@kavithaenair  ohk :)

**Comment by singhpratyush** [comment]
Please look into these points - 

1. Clearing query word after plotting is done

2. Center aligning the loading indicator for Tweets
screenshot from 2017-07-01 11 14 50

3. Improve view for tweets for a query on scroll
screenshot from 2017-07-01 11 15 25

4. Wierd stats are shown on map. And not all numbers are shown at once
screenshot from 2017-07-01 11 16 35

5. Searh is performed without `since` but aggregations are requested with `since`.

6. Inconsistent width
loklakapps surge sh-multilineplotter-


7. Inconsistent height for `Provide Tweet query` and `Search Record` section.

**Comment by djmgit** [comment]
@singhpratyush thanks for pointing these out, I am working on these :)

File: MultiLinePlotter/scripts/app.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+        $scope.data = $scope.data.filter(function(item) {
+            return item[record] === undefined;
+        });
+
+        $scope.plotGraph();
+    }
+
+    // Function to show all the tweets containing the given query word
+
```
Comment (vibhcool) [inline MultiLinePlotter/scripts/app.js]: please remove line-space between function and comment (everywhere)

File: MultiLinePlotter/scripts/app.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            return;
+        }
+        if ($scope.queryRecords.indexOf($scope.tweet) !== -1) {
+            $scope.isError = true;
+            $scope.error = "Graph has already been plotted for this query"
+            return;
+        }
+        $scope.graphLoading = true;
+        $http.jsonp('http://api.loklak.org/api/search.json?callback=JSON_CALLBACK&q=' +
+            $scope.tweet + '%20since:2015-12-10&source=cache&count=0&fields=created_at')
```
Comment (vibhcool) [inline MultiLinePlotter/scripts/app.js]: if you remove `since`, even then , this shall work fine

Comment (vibhcool) [inline MultiLinePlotter/scripts/app.js]: and IMHO dealing with request body will be better approach than dealing with strings.
though this same approach I have to implement in my project 😅

File: MultiLinePlotter/scripts/app.js (hunk 8 above)
Comment (djmgit) [inline MultiLinePlotter/scripts/app.js]: yeah sure :)

**Comment by djmgit** [comment]
Test link : http://loklakapps.surge.sh/MultiLinePlotter/

@singhpratyush @vibhcool  made the changes :)

**Comment by djmgit** [comment]
@singhpratyush did the changes

**Pull request merged**